internal/types/path.go

```diff
@@ -17,17 +17,87 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+// Path should be stored and passed as value. All of its methods return new values, not modifying the receiver's state.
+type Path struct {
+	s []string
+}
+
+// NewPath returns Path from a strings slice.
+func NewPath(path []string) Path {
+	p := Path{s: make([]string, len(path))}
+	copy(p.s, path)
+	return p
+}
+
+// NewPathFromString returns Path from path string. Path string should contain fields separated with '.'.
+func NewPathFromString(s string) Path {
+	path := strings.Split(s, ".")
+
+	return NewPath(path)
+}
+
+// String returns dot-separated path value.
+func (p Path) String() string {
+	return strings.Join(p.s, ".")
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path values array.
+func (p Path) Slice() []string {
```

Comment:
Not obvious to me, that slice returns a slice (sic!).  Maybe rename it to `ToSlice`?

Reply:
`String()`, not `ToString()`. The same logic there